apps/veil/src/pages/tournament/ui/lp-rewards.tsx

```diff
@@ -8,21 +7,133 @@ import { TableCell } from '@penumbra-zone/ui/TableCell';
 import { ValueViewComponent } from '@penumbra-zone/ui/ValueView';
 import { Button } from '@penumbra-zone/ui/Button';
 import { Density } from '@penumbra-zone/ui/Density';
-import { useLpRewards, BASE_LIMIT, BASE_PAGE, Reward } from '../api/use-lp-rewards';
-import { useSortableTableHeaders } from './sortable-table-header';
+import { withdrawPositions } from '@/entities/position/api/withdraw-positions';
+import {
+  useLpRewards,
+  BASE_LIMIT,
+  BASE_PAGE,
+  LpReward,
+} from '@/pages/tournament/api/use-lp-rewards';
+import { useSortableTableHeaders } from '@/pages/tournament/ui/sortable-table-header';
+import { connectionStore } from '@/shared/model/connection';
+import { useRouter } from 'next/navigation';
+import { LpRewardsSortKey } from '../server/lp-rewards';
+import { ValueView, Metadata } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
+import { useAssets } from '@/shared/api/assets';
+import { pnum } from '@penumbra-zone/types/pnum';
+
+function LpRewardRow({
+  lpReward,
+  loading,
+  umMetadata,
+}: {
+  lpReward: LpReward;
+  loading: boolean;
+  umMetadata: Metadata | undefined;
+}) {
+  const router = useRouter();
+  const [isHoveringWithdraw, setIsHoveringWithdraw] = useState(false);
+
+  return (
+    <div
+      onClick={() => {
+        if (!isHoveringWithdraw) {
+          router.push(`/inspect/lp/${bech32mPositionId(lpReward.positionId)}`);
+        }
+      }}
```

Comment:
Instead you should use `event.stopPropagation` on the withdraw button, much simpler than this.